symfony/client/src/core/util/helper/datefns.ts

```diff
@@ -182,6 +293,31 @@ const setClockInterval = (callback: (args: void) => void, interval = 1000) => {
   timer();
 };
 
+/**
+ * guessTimezone will first try to guess the current timezone name using
+ * ES6 Intl API. in the offchance it's not possible it will revert value using
+ * default timezone list.
+ * @typedef {Object} Timezone
+ * @property {string} label - timezone's english name
+ * @property {number} offset - timezone's offset in hours
+ */
+const guessTimezone = () => {
```

Comment:
@samanthajayasinghe @RajithaKumara this function will try to guess the timezone name. as a fallback i've added some hardcoded timezone names as in 4x.